core/integration/cache_test.go

```diff
@@ -107,3 +107,98 @@ func (CacheSuite) TestLocalImportCacheReuse(ctx context.Context, t *testctx.T) {
 
 	require.Equal(t, out1, out2)
 }
+
+func (CacheSuite) TestCacheIsNamespaced(ctx context.Context, t *testctx.T) {
```

Comment:
We'll want an additional test where a function in one module passes a cache volume to a function in a different module and make sure that it works as expected (cache volume should be tied to the one who created it, not used it)